File: Aman/Tree/Find a Corresponding Node of a Binary Tree in a Clone of That Tree.cpp
class Solution {
public:
    TreeNode* res;
    TreeNode* getTargetCopy(TreeNode* org, TreeNode* cl, TreeNode* tar) {
        if(org==NULL)
            return cl;
        if(org==tar)
        {
            res=cl;
            return res;
        }
        getTargetCopy(org->left, cl->left, tar);
        getTargetCopy(org->right, cl->right, tar);
        return res;
    }
};

File: Rohit Singhal/Trees/FindaCorrespondingNodeofaBinaryTreeinaCloneofThatTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {
    
    public TreeNode dfs(TreeNode root,int val)
    {
        if(root==null)
            return null;
        if(root.val==val)
            return root;
        
        TreeNode left=dfs(root.left,val);
        TreeNode right=dfs(root.right,val);
        
        return (left==null) ? right : left;
    }
    
    public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
        return dfs(cloned,target.val);
        
        
        
    }
}
